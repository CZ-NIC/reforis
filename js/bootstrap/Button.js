/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';

export default function Button({icon, className, children, ...props}) {
    className = className ? 'btn ' + className : 'btn btn-primary';

    if (icon)
        icon = <span className={`fa fa-${icon}`}/>;

    return <button className={className} {...props}> {icon} {children} </button>;
}
