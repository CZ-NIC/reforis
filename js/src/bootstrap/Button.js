/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';

const OFFSET = 8;
const SIZE = 2;
const SIZE_CLASS = ' offset-' + OFFSET + ' col-sm-' + SIZE;

export default function Button({className, loading, children, ...props}) {
    className = className ? 'btn ' + className : 'btn btn-primary ';
    className += SIZE_CLASS;

    const span = loading ?
        <span className='spinner-border spinner-border-sm' role='status' aria-hidden='true'/> : null;

    return <button className={className} {...props}>
        {span} {span ? ' ' : null} {children}
    </button>;
}
