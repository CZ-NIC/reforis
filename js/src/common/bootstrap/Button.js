/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import PropTypes from 'prop-types';

const OFFSET = 8;
const SIZE = 3;
const SIZE_CLASS = ' offset-lg-' + OFFSET + ' col-lg-' + SIZE;
const SIZE_CLASS_SM = ' col-sm-12';

Button.propTypes = {
    /** Additional class name. */
    className: PropTypes.string,
    /** Use foris form size and offset. */
    forisFormSize: PropTypes.bool,
    /** Show loading icon. */
    loading: PropTypes.bool,
    /** Button content. */
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.element,
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node),
    ]).isRequired
};

export default function Button({className, loading, forisFormSize, children, ...props}) {
    className = className ? 'btn ' + className : 'btn btn-primary ';
    if (forisFormSize)
        className += SIZE_CLASS + SIZE_CLASS_SM;

    const span = loading ?
        <span className='spinner-border spinner-border-sm' role='status' aria-hidden='true'/> : null;

    return <button className={className} {...props}>
        {span} {span ? ' ' : null} {children}
    </button>;
}
