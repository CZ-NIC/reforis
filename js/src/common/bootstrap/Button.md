Bootstrap button component.

All additional `props` are passed to the `<button>` HTML component.

Can be used without parameters:

```jsx
<Button>Click</Button>
```

Using loading spinner:

```jsx
<Button loading disabled>Loading...</Button>
```
