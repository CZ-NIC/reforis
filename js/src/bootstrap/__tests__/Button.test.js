/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import renderer from 'react-test-renderer';
import React from 'react';
import Button from '../Button'

describe('<Button />', () => {
    it('Render button correctly', () => {
        const tree = renderer
            .create(<Button>Test Button</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Render button with custom classes', () => {
        const tree = renderer
            .create(<Button className="one two three">Test Button</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Render button with spinner', () => {
        const tree = renderer
            .create(<Button loading={true}>Test Button</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});