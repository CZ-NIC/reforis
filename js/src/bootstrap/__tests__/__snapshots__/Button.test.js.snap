// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render button correctly 1`] = `
<button
  class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
>
   
   
  Test Button
</button>
`;

exports[`<Button /> Render button with custom classes 1`] = `
<button
  class="btn one two three offset-lg-8 col-lg-3 col-sm-12"
>
   
   
  Test Button
</button>
`;

exports[`<Button /> Render button with spinner 1`] = `
<button
  class="btn btn-primary  offset-lg-8 col-lg-3 col-sm-12"
>
  <span
    aria-hidden="true"
    class="spinner-border spinner-border-sm"
    role="status"
  />
   
   
   
  Test Button
</button>
`;
